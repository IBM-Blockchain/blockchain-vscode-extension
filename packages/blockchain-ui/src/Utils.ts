import IVSCode from './interfaces/IVSCode';
declare const vscode: IVSCode;

const Utils: any = {

    changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    },

    postToVSCode(message: {command: string, data?: any}): void {
        vscode.postMessage(message);
    }

};

export default Utils;
