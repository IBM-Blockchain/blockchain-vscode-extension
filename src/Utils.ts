const Utils: any = {

    changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

};

export default Utils;
