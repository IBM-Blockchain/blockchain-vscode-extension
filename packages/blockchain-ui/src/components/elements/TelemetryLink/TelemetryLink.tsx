import React, { Component } from 'react';
import { Link } from 'carbon-components-react';
import Utils from '../../../Utils';

interface IProps {
    linkContents: string;
    url: string;
    className?: string;
    id?: string;
}

class TelemetryLink extends Component<IProps> {
    constructor(props: Readonly<IProps>) {
        super(props);
        this.sendTelemetryAndOpen = this.sendTelemetryAndOpen.bind(this);
    }

    sendTelemetryAndOpen(): void {
        Utils.postToVSCode({
            command: 'telemetry',
            data: this.props.linkContents
        });
    }

    render(): JSX.Element {
        return (
            <Link className={this.props.className} href={this.props.url} onClick={this.sendTelemetryAndOpen} id={this.props.id}>
                {this.props.linkContents}
            </Link>
        );
    }
}

export default TelemetryLink;
