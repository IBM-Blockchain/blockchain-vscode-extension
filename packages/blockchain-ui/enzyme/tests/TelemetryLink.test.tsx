import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import TelemetryLink from '../../src/components/elements/TelemetryLink/TelemetryLink';
import Utils from '../../src/Utils';

chai.should();
chai.use(sinonChai);

describe('TelemetryLink component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postToVSCodeStub: sinon.SinonStub;

    beforeEach(() => {
        mySandBox = sinon.createSandbox();
        postToVSCodeStub = mySandBox.stub(Utils, 'postToVSCode').resolves();
    });

    afterEach(() => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<TelemetryLink linkContents='my link' url='www.someurl.com'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should send telemetry information to VS Code when clicked', () => {
        const component: any = mount(<TelemetryLink linkContents='my link' url='www.someurl.com'/>);
        component.find('a').simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: 'telemetry',
            data: 'my link'
        });
    });

    it('should add any adidtional styles provided', () => {
        const component: any = mount(<TelemetryLink linkContents='my link' url='www.someurl.com' className='some-style'/>);
        component.find('a').hasClass('some-style').should.equal(true);
    });
});
