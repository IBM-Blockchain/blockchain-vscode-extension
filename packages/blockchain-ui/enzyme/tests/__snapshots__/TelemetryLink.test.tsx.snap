// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelemetryLink component should render the expected snapshot 1`] = `
<a
  className="bx--link"
  href="www.someurl.com"
  onClick={[Function]}
>
  my link
</a>
`;
